### ISSUE TITLE
[a11y]: User must press TAB to focus before pressing RETURN to apply ordering on datagrid columns

### ISSUE THREAD

**Issue by mconway-ns1** [carbon-design-system/ibm-products#5079]
### Package

@carbon/ibm-products

### Browser

Chrome, Firefox

### Operating System

MacOS

### Package version

2.36.0

### React version

18.2.0 

### Automated testing tool and ruleset

Manual

### Assistive technology

Macos

### Description

Datagrid sort loses focus after a sort,  forcing a user using keyboard navigation to {TAB} and press {ENTER} again to sort in the opposite direction instead of just pressing {ENTER} 
User should not have to TAB on the focus state so it should act where you TAB and can press ENTER to sort and press ENTER again to sort another direction or TAB to jump to the next column.

### WCAG 2.1 Violation

SC 2.4.3 Focus order

### Reproduction/example

https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-datagrid--sortable-columns

### Steps to reproduce

Steps to reproduce
1. Navigate to a data grid 
2. Use TAB to navigate to the column.
3. Press RETURN to sort column.
4. Press RETURN to sort column in opposite direction 

Expected 
sort in opposite direct

Actual

https://github.com/carbon-design-system/ibm-products/assets/156674431/ed12c466-75cf-496d-8b6f-a3dcade23d16

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
